osi_object.proto

```diff
@@ -512,6 +512,191 @@ message MovingObject
         TYPE_ANIMAL = 4;
     }
 
+    // Specific information about the pedestrian.
+    //
+    // \note This field is mandatory if the \c #type is
+    // #TYPE_PEDESTRIAN.
+    //
+    // \rules
+    // check_if this.type is_equal_to 3 else do_check is_set
+    // \endrules
+    //
+    optional PedestrianAttributes pedestrian_attributes = 12;
+    
+    //
+    // \brief Additional pedestrian data in case the moving object is a human.
+    //
+    // This is an extension to the \c MovingObject with additional information 
+    // describing a pedestrian in more detail.
+    // 
+    message PedestrianAttributes {
+
+        // Position offset from the center of the bounding box to the current position
+        // of the root point of the pedestrian skeleton model.
+        // 
+        optional Vector3d bbcenter_to_root = 1;
+
+        // List of all bones of the pedestrian
+        // 
+        // The number of bones may vary, based on the detail level of 
+        // the pedestrian model used. For example, some simulators will not include 
+        // detailed data about the hands of a pedestrian.
+        // 
+        repeated Bone skeleton_data = 2;
+         
+        // 
+        // \brief Bones in the skeleton of the pedestrian
+        //
+        // Each point represents a joint, or otherwise important point in the skeleton 
+        // of a pedestrian. For example pelvis, knee or shoulder. The naming convention 
+        // should be followed for identifying bones.
+        //
+        message Bone {
+
+            // Bones are identified by their type, combined with which body side
+            // they are on.
+            //  
+            // To properly identify the bones the pre-defined naming convention 
+            // must be used.
+            //
+            // \image html OSI_SkeletonNamingConvention.svg
+            //
+            // A bone's name, position, and orientation is defined by the end-point closer
+            // to the root. For example: the "LOWER_ARM_L" will define the point in the left
+            // elbow of the model.
+            //
+            // If a bone which is more than one layer detached from the
+            // root point is used, all bones between that bone and the root also
+            // need to be defined in order to create a complete chain!
+            //
+            // If information about bones are missing, they may be left empty. A complete
```

Comment:
Short-notice review: I'd remove "A complete set of bones for the entire skeleton cant't be expected!". This is potentially misleading.